packages/store/src/StoreCore.sol

```diff
@@ -74,7 +75,11 @@ library StoreCore {
    * Get the schema for the given table
    */
   function getSchema(uint256 table) internal view returns (Schema schema) {
-    return StoreCoreInternal._getSchema(table);
+    schema = StoreCoreInternal._getSchema(table);
+    // A race condition prevents us from doing this check ~everywhere
+    // if (schema.isEmpty()) {
+    //   revert StoreCore_TableNotFound(table);
+    // }
```

Comment:
I saw we were doing this check in just one of the read functions (removed below) but not in the others or in any of the writes. I thought this would be a cheap check to do and worth adding everywhere, to make the error messages more useful (was getting `StoreCore_InvalidDataLength` rather than `StoreCore_TableNotFound` in tests).

But, there seems to be a race condition somewhere when registering schemas and/or metadata, where the schema is requested, but not found, therefore we can't write to it. So I think somewhere we're sort of writing data ahead of the schema existing?

Reply:
Just did a bit more testing, and this issue comes from a `getField` call.

Reply:
<img width="485" alt="image" src="https://user-images.githubusercontent.com/508855/223525218-a15eeaff-0cea-40af-978e-6e27e528b8a1.png">

```

Running 1 test for test/World.t.sol:WorldTest
[FAIL. Reason: Setup failed: Schema not found in getField call] setUp() (gas: 0)
Traces:
  [120595] WorldTest::setUp()
    ├─ [79789] → new <Unknown>@0xFEfC6BAF87cF3684058D62Da40Ff3A795946Ab06
    │   ├─ emit StoreSetRecord(table: 31851884016513291979465943704761379125824557917914706133157551191634376234846, key: [0x466b87090ce2bc34362737dab7b900dfc94a202aee6a6323f6813f7ce59d975e], data: 0x002001005f000000000000000000000000000000000000000000000000000000)
    │   ├─ emit StoreSetRecord(table: 31851884016513291979465943704761379125824557917914706133157551191634376234846, key: [0x15985d03d398bf6f1a658e500e78fbbf35cd784646b3998d4f26a9a9e441a627], data: 0x00000002c5c40000000000000000000000000000000000000000000000000000)
    │   ├─ emit StoreSetRecord(table: 9767772539668730435636382986942077369059354384034778394590643984438732629543, key: [0x15985d03d398bf6f1a658e500e78fbbf35cd784646b3998d4f26a9a9e441a627], data: 0x0000010d000d010000000000000000000000000000000000000000000000000053746f72654d65746164617461000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000008000000000000000000000000000000000000000000000000000000000000000097461626c654e616d6500000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000014616269456e636f6465644669656c644e616d6573000000000000000000000000)
    │   └─ ← 0 bytes of code
    └─ ← "Schema not found in getField call"
```

Reply:
`StoreMetadata.registerSchema();` calls `StoreSwitch.setMetadata(_tableId, _tableName, _fieldNames);`, which tries to call 
`getField` for hooks when setting metadata fields, but the hook table isn't registered yet
And if you do hooks table first the issue is reversed for metadata